### PR TITLE
Bugs: Fix more invalid rules

### PR DESCRIPTION
**Pull request by Dorian**
- [x] Fix specs

(A fuzzer could be better at this than me :D )

### CODE REVIEW

File: lib/ice_cube/rule.rb [hunk 1]
```diff
@@ -73,6 +73,7 @@ class << self
       # Convert from a hash and create a rule
       def from_hash(original_hash)
         hash = IceCube::FlexibleHash.new original_hash
+        raise ArgumentError, 'empty ical rule_type' if hash[:rule_type].nil?
```
Comment (seejohnrun) [inline lib/ice_cube/rule.rb]: I think we can do something more here by saying:

`raise ArgumentError, 'Invalid rule type' unless hash[:rule_type] && hash[:rule_type].match(/\:\:(.+?)Rule/)`

File: lib/ice_cube/parsers/ical_parser.rb [hunk 2]
```diff
@@ -31,6 +31,7 @@ def self.rule_from_ical(ical)
 
       ical.split(';').each do |rule|
         (name, value) = rule.split('=')
+        raise ArgumentError, "empty ical rule's value" if value.nil?
```
Comment (seejohnrun) [inline lib/ice_cube/parsers/ical_parser.rb]: I think the text of this error should be something more like: "Invalid iCal rule component"

**Comment by Dorian** [comment]
@seejohnrun What about now?

**Comment by seejohnrun** [comment]
👍

**Pull request merged**